bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,326 @@
+= WFLY-13305: WildFly bootable JAR
+:author:            Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We are proposing a way to package server and application inside a bootable JAR. The JAR contains everything needed to run the server.
+The packaged server can benefit from Galleon tooling and be trimmed down to user needs (thanks to Galleon layers).
+Such executable JAR can then be run on bare metal and in cloud execution contexts.
+
+A simple example of a https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/jaxrs/pom.xml[pom file] to create a WildFly bootable JAR.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13305[WFLY-13305]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1385[EAP7-1385]
+
+* https://issues.redhat.com/browse/WFCORE-4895[WFCORE-4895]
+
+* https://issues.redhat.com/browse/WFGP-169[WFGP-169]
+
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven plugins]
+
+* https://github.com/wildfly/quickstart[WildFly Quickstarts]
+
+* https://start.microprofile.io/[Microprofile starter]
+
+* https://thorntail.io/[Thorntail]
+
+== Requirements
+
+==== Bootable JAR server vs Standalone server
+
+Although the bootable JAR can be seen as a standalone server the following limitations apply:
+
+* CLI management operations that would imply a server restart are not supported.
+* :shutdown operation with restart (e.g.: by sending CLI operation :shutdown(restart=true)) is not supported. 
+  A warning is displayed and the bootable JAR process is terminated.
+* Changes done to the management model (e.g.: using CLI or Web console) are not persisted between bootable JAR executions.
```

Comment:
Would it be possible to mention to the fact that instead CLI script execution during the server packaging could be used as a mechanism to configure the server in a way that would survive restarts? See chapter #6 in current community documentation. It seems quite important to me from the user experience point of view. WDYT?

Reply:
Ok clarified.